### ISSUE TITLE
request for assistance: configuring the intro.lua in neovim

### ISSUE THREAD

**Issue by bassamsdata** [Bekaboo/nvim#2]
firstly, i want to express my gratitude for the fantastic setup you've provided—it's been truly inspiring.

i'm currently attempting to replace my `mini.starter` plugin with your minimalist intro message, aiming for a sleeker interface with just a concise message. however, i've encountered an issue. the file `plugin/intro.lua` doesn't seem to be referenced anywhere in your configuration. despite this, neovim appears to require the file during startup. i also checked the profiling of lazy loading but couldn't discern how it might be related.

I did require the file in my `init.lua`, but nothing happens.

I greatly appreciate your assistance. 

thank you!

**Comment by Bekaboo** [comment]
Hi, thanks for your feedback.

Vim/Neovim has a builtin mechanism to load plugins at startup without the need to explicitly `require` or `:source`, see `:h runtimepath`. TLDR, files under `plugin/` are automatically loaded during startup. It should just work if you put `intro.lua` under your `~/.conifg/nvim/plugin/` directory.

---

EDIT: If you are going to use `intro.lua` in your config you may also want to change these two lines:

https://github.com/Bekaboo/nvim/blob/26e8d500e7e219fd686938adf193d9d9a29d2dad/plugin/intro.lua#L8

https://github.com/Bekaboo/nvim/blob/26e8d500e7e219fd686938adf193d9d9a29d2dad/plugin/intro.lua#L4

by removing the checkings for `vim.g.has_ui` and `vim.g.modern_ui`, they are global vim variables set at startup in [core/general.lua](https://github.com/Bekaboo/nvim/blob/master/lua/core/general.lua) here:

https://github.com/Bekaboo/nvim/blob/26e8d500e7e219fd686938adf193d9d9a29d2dad/lua/core/general.lua#L8

and are used in my config to indicate:

- if neovim has UI attached (for example, if neovim is launched with `--headless` then it will not have UI), and
- if neovim can render modern UI (all unicode characters, nerd fonts, etc), not possible when `$DISPLAY` is empty, which means that we are in linux tty

I use these two variables to disable some UI plugins when neovim has no UI and render differently in tty to have a more consistent and usable appearance in tty.

So I suggest either you set these variables the same way as I do before loading `intro.lua` or simply ignore these variables, else `intro.lua` will refuse to show intro messages or show an inferior (ASCII code only) message because it thinks there's no UI attached or neovim is launched in tty instead of in a modern terminal emulator.

**Comment by bassamsdata** [comment]
Thank you for your thorough response – much appreciated!

surprisingly, I wasn't aware that simply creating a directory with the `plugin` name would lead neovim to load it directly. that's a neat feature, and i'm grateful for the insight.

although i've made progress, the issue persists. neovim now detects the file, and it appears in the lazy profile. i removed all references to `has_ui` and `modern_ui`, but unfortunately, the problem persists. interestingly, deleting the line related to `vim.opt.shortmess` option results in the default intro showing up, indicating that neovim is indeed applying it.

I noticed that after making an error in the options (vim.opt), an error on neovim startup triggers the customized intro to appear 🫣 (but without the colorscheme due to the 'vim.opt' error).

i'll do some debugging to pinpoint the issue further. your assistance has been invaluable. thanks again!

**Comment by Bekaboo** [comment]
What is the output of `nvim --version`?

For debugging I suggest creating directory `~/.config/nvim-test-foo-bar/plugin`, putting `intro.lua` under that directory, launching nvim using `NVIM_APPNAME=nvim-test-foo-bar nvim` and see if the intro message shows up.

**Comment by bassamsdata** [comment]
I've been testing on stable v0.9.4 and 
NVIM v0.10.0-dev-1744+g0d885247b

On stable, the no config, `NVIM_APPNAME=nvim-test-foo-bar nvim` worked smoothly but didn't work on nightly.

since neovim detect the file, I thought the problem might be in the clear autocmd (last one in intro.lua), so I've been testing it and found if I removed this event  `"CursorMoved"` the intro worked smoothly on nightly and stable and in my own config. (I have a backup config, I had to delete `"InsertEnter"` event as well to be able to work). it's weird but idk why.

thanks a lot
